client-local/src/main/java/com/cloudera/livy/client/local/LocalClient.java

```diff
@@ -339,21 +326,14 @@ public void run() {
       if (userJars != null) {
         String allJars = Joiner.on(",").join(livyJars, userJars);
         conf.set(SPARK_JARS_KEY, allJars);
-      } else {
-        argv.add("--jars");
-        argv.add(livyJars);
       }
 
-      argv.add(jar);
-      argv.add("--remote-host");
-      argv.add(serverAddress);
-      argv.add("--remote-port");
-      argv.add(serverPort);
-      argv.add("--config-file");
-      argv.add(livyConf.getAbsolutePath());
+      launcher.addAppArgs("--remote-host" + serverAddress);
```

Comment:
As I mentioned earlier, this is not how you add multiple app args. Use:

    launcher.addAppArgs("--remote-host, serverAddress);